yage-gl/src/objects/buffer.rs

```diff
@@ -84,10 +62,106 @@ impl<'a> Buffer<'a> {
     pub fn attrib_disable(&self, index: u32) {
         self.gl.disable_vertex_attrib_array(index);
     }
+
+    /// Bind the buffer and return BufferUpdater that can be used for updating
+    ///
+    /// # Example
+    /// ```ignore
+    /// let buffer = Buffer::new(gl);
+    /// buffer.update()
+    ///     .set_data(...)
+    ///     .attrib_enable(...)
+    /// ```
+    ///
+    /// Caution: assumes that you do nothing between calls to the updater that would
+    /// unbind it!
+    pub fn update(&self) -> BufferUpdater {
+        // TODO!: boolean param for bind also here? (-> without would be simpler)
+        BufferUpdater::new(self, true)
+    }
 }
 
 impl<'a> Drop for Buffer<'a> {
     fn drop(&mut self) {
         self.gl.delete_buffer(&self.buffer_handle);
     }
 }
+
+pub struct BufferUpdater<'a> {
+    buffer: &'a Buffer<'a>,
+}
+
+/// Builder-like object for updating without multiple binds.
+impl<'a> BufferUpdater<'a> {
+    /// It is expected that this is usually only constructed via Buffer::update().
+    /// For exceptional cases where binding is not desired (already bound),
+    /// this constructor can be used with `bind` set to `false`.
+    pub fn new(buffer: &'a Buffer, bind: bool) -> Self {
+        if bind {
+            buffer.bind()
+        }
+        Self { buffer }
+    }
+
+    // TODO!!: self vs &self return types??
```

Comment:
There doesn't seem to be a clear consensus what is better...but `rust-derive-builder` switched to references, so we'll keep it as references here too for now.